### PR TITLE
Switch policy-agent to qrexec

### PR DESCRIPTION
**Pull request by pwmarcz**
Requires QubesOS/qubes-core-admin#323.

### CODE REVIEW

**Comment by marmarek** [comment]
[Pylint complains](https://travis-ci.com/QubesOS/qubes-core-qrexec/jobs/291030166):
```
************* Module qrexec.server
qrexec/server.py:14:-1: W0105: String statement has no effect (pointless-string-statement)
************* Module qrexec.tools.qrexec_policy_agent
qrexec/tools/qrexec_policy_agent.py:521:44: E0602: Undefined variable 'rpc_path' (undefined-variable)
qrexec/tools/qrexec_policy_agent.py:516:4: W0612: Unused variable 'args' (unused-variable)
qrexec/tools/qrexec_policy_agent.py:39:0: C0411: third party import "import gbulb" should be placed before "import gi" (wrong-import-order)
```

File: qrexec/server.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+A client and server for socket-based QubesRPC (currently used for
+qrexec-policy-agent).
+
+The request (dom0 -> VM) is JSON-encoded, response is plain ASCII text.
+'''
+
+
+async def client_connected(reader, writer, handler):
+    try:
+        first_line = await reader.readline()
```
Comment (marmarek) [inline qrexec/server.py]: Service header is `\0` terminated, not `\n` terminated.

File: qrexec/server.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        return call_socket_service_local(
+            service, source_domain, params, rpc_path)
+    return call_socket_service_remote(
+        remote_domain, service, source_domain, params)
+
+async def call_socket_service_local(service, source_domain, params,
+                                    rpc_path=RPC_PATH):
+    path = os.path.join(rpc_path, service)
+    reader, writer = await asyncio.open_unix_connection(path)
+    writer.write('{} {}\n'.format(service, source_domain).encode('ascii'))
```
Comment (marmarek) [inline qrexec/server.py]: And here too.

**Comment by pwmarcz** [comment]
This should be done except for the Debian package for gbulb.

**Comment by pwmarcz** [comment]
Superseded by #30 that refactors this code further.

**Pull request closed without merging**